geospace-lib/src/main/scala/com/socrata/geospace/lib/shapefile/ShapeFileReader.scala

```diff
@@ -0,0 +1,46 @@
+package com.socrata.geospace.lib.shapefile
+
+import java.io.{IOException, File}
+
+import com.socrata.geospace.lib.errors.InvalidShapefileSet
+import org.geoscript.projection._
+import org.geotools.factory.Hints
+import org.geotools.referencing.ReferencingFactoryFinder
+
+
+trait ShapeReader{
+  def read(file: File): AnyRef
+  def validate(file: File): AnyRef
+
+  /** Provides the projection object necessary to re-project when parsing shape-files **/
+  def getTargetProjection(epsgCode: String, forceLonLat: Boolean): Either[InvalidShapefileSet, Projection] =
+    try {
+      val hints = if (forceLonLat) new Hints(Hints.FORCE_LONGITUDE_FIRST_AXIS_ORDER, true) else new Hints()
+      val factory = ReferencingFactoryFinder.getCRSAuthorityFactory("EPSG", hints)
+      Right(factory.createCoordinateReferenceSystem(epsgCode))
+    } catch {
+      case e: Exception if !e.isInstanceOf[IOException] =>
```

Comment:
I think what you want here is just to catch the `NoSuchAuthorityCodeException` and leave other exceptions unhandled. Eg:
```scala
case notFound: NoSuchAuthorityCodeException => 
  Left(InvalidShapefileSet("Unable to find target projection: " + notFound.getMessage))
```